### PR TITLE
feat: deploy storybook to vercel

### PR DESCRIPTION
**Pull request by mikasackermn**
# Summary

Deploy Storybook to Vercel

In order to deploy Storybook in Vercel, we made the decision to detach it from the user interface and create a new Storybook package.

### CODE REVIEW

**Review by nikaaru — APPROVED** [review]
LGTM

File: widget/storybook/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
+  "extends": "../../tsconfig.lib.json",
+  "include": ["src", "types"],
+  "compilerOptions": {
+    "outDir": "storybook-static",
+    "lib": ["dom", "esnext"],
+    // match output dir to input dir. e.g. dist/index instead of dist/src/index
+    "baseUrl": ".",
+    "rootDirs": ["./src", "../../../translations"],
```
Comment (yeager-eren) [inline widget/storybook/tsconfig.json:10]: Do we need `/translations` here?

**Review by yeager-eren — CHANGES_REQUESTED** [review]

**Comment by yeager-eren** [comment]
It's not related to your PR but please change `Conventional Commits Check` to `Check PR` (`.github/workflows/checks.yaml:1`)

File: widget/storybook/package.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  "files": [
+    "storybook-static",
+    "src"
+  ],
+  "scripts": {
+    "dev": "storybook dev -p 3000",
+    "build": "storybook build && cp ./vercel.json ./storybook-static",
+    "ts-check": "tsc --declaration --emitDeclarationOnly -p ./tsconfig.json",
+    "type-checking": "tsc --declaration --emitDeclarationOnly",
+    "clean": "rimraf dist",
```
Comment (yeager-eren) [inline widget/storybook/package.json:17]: this should be `storybook-static` in your case

File: widget/storybook/package.json [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    "@storybook/react-webpack5": "^8.0.8",
+    "@types/react": "^18.0.25",
+    "@types/react-dom": "^18.0.8",
+    "babel-loader": "^9.1.0",
+    "husky": "^8.0.2",
+    "react": "^18.2.0",
+    "react-is": "^18.2.0",
+    "storybook": "^8.0.8"
+  },
+  "dependencies": {
```
Comment (yeager-eren) [inline widget/storybook/package.json:45]: Seems dependencies are copy pasted from `ui`, please make sure keeping those that neede. (same applies to devDependencies)

File: widget/storybook/readme.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# @rango-dev/storybook
+
+Rango UI design system preview
```
Comment (yeager-eren) [inline widget/storybook/readme.md:3]: Storybook for Rango design system

**Comment by yeager-eren** [comment]
please also make sure `yarn dev:ui` will run your new package.

File: widget/storybook/package.json (hunk 2 above)
Comment (mikasackermn) [reply widget/storybook/package.json:17]: This folder is created with yarn build

Comment (yeager-eren) [reply widget/storybook/package.json:17]: you are removing `dist` which doesn't exist here. `yarn build` is creating a folder called `storybook-static`. so here shoudl be `rimraf storybook-static`

File: widget/storybook/package.json (hunk 3 above)
Comment (yeager-eren) [reply widget/storybook/package.json:40]: You missed to remove `husky` (you can remove from `/ui` as well, it should be there as well).
And also `dependency` and `devDependencies` is not correct. please change it to something like this:
```json
  "devDependencies": {
    "@types/react": "^18.0.25",
    "@types/react-dom": "^18.0.8",
  },
  "dependencies": {
     "@chromatic-com/storybook": "^1",
    "@storybook/addon-essentials": "^8.0.8",
    "@storybook/addon-info": "^5.3.21",
    "@storybook/addon-links": "^8.0.8",
    "@storybook/addons": "^7.1.1",
    "@storybook/react": "^8.0.8",
    "@storybook/react-vite": "^8.0.8",
    "@storybook/react-webpack5": "^8.0.8",
    "@rango-dev/ui": "^0.31.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
     "react-is": "^18.2.0",
    "storybook": "^8.0.8"
  },
```

Comment (yeager-eren) [reply widget/storybook/package.json:40]: I also not sure why `@chromatic-com/storybook` has been added. Do we need that?

File: package.json [hunk 5]
```diff
@@ -27,7 +27,9 @@
     "build:all": "nx run-many --target=build",
     "build": "nx affected --target=build --parallel=3",
     "build:non-demo": "nx affected --target=build --parallel=3 --exclude=@rango-dev/queue-manager-demo,@rango-dev/wallets-demo,@rango-dev/wallets-adapter-demo,@rango-dev/widget-app,@rango-dev/widget-playground",
-    "dev:ui": "nx run-many --target=storybook --projects=@rango-dev/ui",
+    "dev:ui:type-checker": "node ./scripts/dev-ts-check/command.mjs --project @rango-dev/storybook",
+    "dev:ui:dev-server": "yarn --cwd ./widget/storybook/ dev",
+    "dev:ui": "npm-run-all --parallel dev:ui:dev-server dev:ui:type-checker",
```
Comment (yeager-eren) [inline package.json:32]: please use this command instead and remove the other ones as well:
```
"dev:ui": "nx run @rango-dev/storybook:dev",
```

Running using nx will build necessary packages as well.

**Review by yeager-eren — APPROVED** [review]
lgtm

**Pull request merged**